lib/tenstorrent/bh_chip/bh_chip.c

```diff
@@ -132,7 +133,51 @@ int therm_trip_gpio_setup(struct bh_chip *chip)
 	}
 	gpio_init_callback(&chip->therm_trip_cb, therm_trip_detected,
 			   BIT(chip->config.therm_trip.pin));
-	ret = gpio_add_callback(chip->config.therm_trip.port, &chip->therm_trip_cb);
+	ret = gpio_add_callback_dt(&chip->config.therm_trip, &chip->therm_trip_cb);
+
+	return ret;
+}
+
+void pgood_fall_detected(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
+{
+	struct bh_chip *chip = CONTAINER_OF(cb, struct bh_chip, pgood_fall_cb);
+
+	chip->data.pgood_fall_triggered = true;
+	bm_event_post(WAKE_BM_MAIN_LOOP);
+}
+
+void pgood_rise_detected(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
+{
+	struct bh_chip *chip = CONTAINER_OF(cb, struct bh_chip, pgood_rise_cb);
+
+	chip->data.pgood_rise_triggered = true;
+	bm_event_post(WAKE_BM_MAIN_LOOP);
+}
+
+int pgood_gpio_setup(struct bh_chip *chip)
+{
+	/* Set up PGOOD interrupt */
+	int ret;
+
+	ret = gpio_pin_configure_dt(&chip->config.pgood, GPIO_INPUT);
+	if (ret != 0) {
+		return ret;
+	}
+	ret = gpio_pin_interrupt_configure_dt(&chip->config.pgood, GPIO_INT_EDGE_TO_INACTIVE);
+	if (ret != 0) {
+		return ret;
+	}
+	gpio_init_callback(&chip->pgood_fall_cb, pgood_fall_detected,
+			   BIT(chip->config.pgood.pin));
+	ret = gpio_add_callback_dt(&chip->config.pgood, &chip->pgood_fall_cb);
+
+	ret = gpio_pin_interrupt_configure_dt(&chip->config.pgood, GPIO_INT_EDGE_TO_ACTIVE);
+	if (ret != 0) {
+		return ret;
+	}
+	gpio_init_callback(&chip->pgood_rise_cb, pgood_rise_detected,
+			   BIT(chip->config.pgood.pin));
+	ret = gpio_add_callback_dt(&chip->config.pgood, &chip->pgood_fall_cb);
```

Comment:
Just thinking it would be a bit safer to initialize callbacks and add them to the driver prior to enabling interrupts. Also, need to configure interrupts once, since the second configuration would overwrite the first.



```suggestion
	LOG_MODULE_REGISTER(bh_chip, CONFIG_TT_BH_CHIP_LOG_LEVEL);
	/* ^^ at the top of this file, and would need a logging entry in lib/tenstorrent/bh_chip/Kconfig */

	gpio_init_callback(&chip->pgood_fall_cb, pgood_fall_detected,
			   BIT(chip->config.pgood.pin));
	ret = gpio_add_callback_dt(&chip->config.pgood, &chip->pgood_fall_cb);
	if (ret != 0) {
		LOG_ERR("%s() failed: %d", "gpio_add_callback_dt", ret);
		return ret;
	}

	gpio_init_callback(&chip->pgood_rise_cb, pgood_rise_detected,
			   BIT(chip->config.pgood.pin));
	ret = gpio_add_callback_dt(&chip->config.pgood, &chip->pgood_fall_cb);
	if (ret != 0) {
		LOG_ERR("%s() failed: %d", "gpio_pin_interrupt_configure_dt", ret);
		return ret;
	}
	
	ret = gpio_pin_interrupt_configure_dt(&chip->config.pgood, GPIO_INT_EDGE_BOTH);
	if (ret != 0) {
		LOG_ERR("%s() failed: %d", "gpio_pin_interrupt_configure_dt", ret);
		return ret;
	}
```